File: examples/hello.go
package main

import "fmt"

func main() {
    fmt.Println("Hello from Go!")
    
    // Variables and types
    var message string = "Go is awesome!"
    number := 42
    
    fmt.Printf("Message: %s\n", message)
    fmt.Printf("Number: %d\n", number)
    
    // Slice
    numbers := []int{1, 2, 3, 4, 5}
    fmt.Println("Numbers:", numbers)
    
    // Function call
    result := add(10, 20)
    fmt.Printf("10 + 20 = %d\n", result)
}

func add(a, b int) int {
    return a + b
}

File: examples/hello.rs
// Rust Example - Safe Systems Programming

fn main() {
    println!("🦀 Hello from Rust!");
    
    // Variables
    let message = "Rust is blazingly fast!";
    let number: i32 = 42;
    
    println!("Message: {}", message);
    println!("Number: {}", number);
    
    // Vector
    let numbers = vec![1, 2, 3, 4, 5];
    println!("Numbers: {:?}", numbers);
    
    // Function call
    let result = add(10, 20);
    println!("10 + 20 = {}", result);
    
    // Pattern matching
    let value = Some(5);
    match value {
        Some(n) => println!("Got value: {}", n),
        None => println!("No value"),
    }
}

fn add(a: i32, b: i32) -> i32 {
    a + b
}
